hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -227,16 +264,47 @@ ServerName regionServerStartup(RegionServerStartupRequest request, int versionNu
     final String hostname =
       request.hasUseThisHostnameInstead() ? request.getUseThisHostnameInstead() : isaHostName;
     ServerName sn = ServerName.valueOf(hostname, request.getPort(), request.getServerStartCode());
+
+    // Check if the HMaster is configured to reject decommissioned hosts or not.
+    // When configured to do so, all hosts of decommissioned RegionServers are prevented from
+    // reporting for duty; otherwise, we accept them unless they come back with an identical
+    // ServerName that we can find in the decommissioned hosts.
+    // See HBASE-28342 for details.
+    if (shouldHostBeRejected(hostname)) {
+      LOG.warn("Rejecting RegionServer {} from reporting for duty because Master is configured "
```

Comment:
I don't think this is a warn level log, info is fine.

Reply:
Fixed.